File: 2026/05-May/06-Wednesday/solution.py
class Solution:

    def rotateTheBox(self, boxGrid: List[List[str]]) -> List[List[str]]:

        m = len(boxGrid)
        n = len(boxGrid[0])

        for row in boxGrid:

            empty_pos = n - 1

            for j in range(n - 1, -1, -1):

                if row[j] == '*':
                    empty_pos = j - 1

                elif row[j] == '#':
                    row[j], row[empty_pos] = row[empty_pos], row[j]
                    empty_pos -= 1

        ans = [[0] * m for _ in range(n)]

        for i in range(m):
            for j in range(n):
                ans[j][m - 1 - i] = boxGrid[i][j]

        return ans

File: 2026/05-May/06-Wednesday/solution.cs
public class Solution {

    public char[][] RotateTheBox(char[][] boxGrid) {

        int m = boxGrid.Length;
        int n = boxGrid[0].Length;

        for (int i = 0; i < m; i++) {

            int emptyPos = n - 1;

            for (int j = n - 1; j >= 0; j--) {

                if (boxGrid[i][j] == '*') {
                    emptyPos = j - 1;
                }

                else if (boxGrid[i][j] == '#') {

                    char temp = boxGrid[i][j];
                    boxGrid[i][j] = boxGrid[i][emptyPos];
                    boxGrid[i][emptyPos] = temp;

                    emptyPos--;
                }
            }
        }

        char[][] ans = new char[n][];

        for (int i = 0; i < n; i++) {
            ans[i] = new char[m];
        }

        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                ans[j][m - 1 - i] = boxGrid[i][j];
            }
        }

        return ans;
    }
}
